Update Stackdriver Logging details

charlesbaer opened the pull request:
Updated the Stackdriver Logging section to include the Cloud Ops for GKE resource model and guidance on how to use.
Updated the options for exports.

<!-- 🛈

 Hello!

 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/start/#improve-existing-content

 Use the default base branch, “master”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.

-->

Arhell reviewed (commented):
Pls sign the CLA in order for the PR to be reviewed.
more info
https://github.com/kubernetes/community/blob/master/CLA.md

RA489 commented:
/check-cla

charlesbaer commented:
The CLA request has been completed

charlesbaer commented:
> I believe we are trying to exclude vendor-specific or dual-hosted contents in the website.

The page title is "Logging Using Stackdriver" which scopes the contents of the docs page to using Stackdriver for Kubernetes logs. In this PR, I updated the incomplete and out-dated references.  As they currently stand, users may be confused when they start using Stackdriver Logging based on the contents of this document. I did not add new content, but rather, corrected or expanded the existing content.

celestehorgan commented:
Preview link: https://deploy-preview-26925--kubernetes-io-master-staging.netlify.app/docs/tasks/debug-application-cluster/logging-stackdriver/#verifying-your-logging-agent-deployment

celestehorgan commented on content/en/docs/tasks/debug-application-cluster/logging-stackdriver.md:
```diff
@@ -211,32 +211,45 @@ Stackdriver Logging agent attaches metadata to each log entry, for you to use la
+the resource type of a container log is `container`, which is named
+`GKE Containers` in the UI. For Cloud Ops for GKE, the resource type of a container
+log is `k8s_container`, which is named `Kubernetes Container` in the UI. The resource
+types are used even if the Kubernetes cluster is not on Google Kubernetes Engine).
```
```suggestion
types are used even if the Kubernetes cluster is not on Google Kubernetes Engine.
```

celestehorgan reviewed (changes requested):
Per the third party content section of the content guide: https://kubernetes.io/docs/contribute/style/content-guide/#third-party-content:

- I agree with @charlesbaer, this is an update of existing documentation and it is necessary for Kubernetes to function.
- While this PR _does_ introduce mention (and link to) Cloud Ops for GKE that wasn't previously present, I think it's necessary, as logging behaviour differs because of it. 

No major changes to this PR, but one typo fix.

celestehorgan commented:
I'd love if someone from @kubernetes/sig-instrumentation-approvers could take a look at this and give a technical LGTM. 

/sig instrumentation

ehashman commented:
@celestehorgan we removed support of stackdriver in our e2e tests, so we may want to remove this page entirely. I'll add to SIG Instrumentation agenda this week to either find a reviewer or make a decision.

celestehorgan commented:
Thanks @ehashman! Loop back round here or in slack and let me know how this goes.

logicalhan commented:
/sig cloud-provider
/cc @cheftako

ehashman commented:
> I believe we are trying to exclude vendor-specific or dual-hosted contents in the website.

SIG Instrumentation is fine with this (and removing this vendor-specific doc), but we think these docs don't belong to us. Pinging SIG Cloud Provider.

tengqm commented:
/hold
So the current consensus is that this page should be removed rather than fixed.

kbhawkey commented:
Related to PR https://github.com/kubernetes/website/pull/28166.

kbhawkey commented:
👋 @charlesbaer .
Since there has been no recent activity on this PR, I plan to close it.
Feel free to reopen this pull request.
/close
